src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -315,13 +320,21 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   auto timeout_point = ToTimeoutPoint(timeout_ms);
   bool locked = false;
   bool expired = false;
-  do {
-    RAY_RETURN_NOT_OK(object->header->CheckHasError());
-    // The channel is still open. This lock ensures that there is only one reader
-    // at a time. The lock is released in `ReadRelease()`.
-    locked = channel->lock->try_lock();
-    expired = timeout_point && std::chrono::steady_clock::now() >= *timeout_point;
-  } while (!locked && !expired);
+  {
+    std::signal(SIGINT, SignalHandler);
```

Comment:
What is the scope of the signal handler? Does it affect the whole process?

Reply:
it's just this scope where it's contained, not the whole process

Reply:
hmm, that sounds surprising. If you call `std::signal(SIGINT, SignalHandler)` inside a function, the setting still takes effect after execution leaves the function, right?